api/src/main/java/org/pmiops/workbench/actionaudit/auditors/UserServiceAuditAdapterImpl.java

```diff
@@ -39,13 +42,29 @@ public UserServiceAuditAdapterImpl(
 
   @Override
   public void fireUpdateDataAccessAction(
-      DbUser targetUser, DataAccessLevel dataAccessLevel, DataAccessLevel previousDataAccessLevel) {
+      DbUser targetUser,
+      DataAccessLevel dataAccessLevel,
+      DataAccessLevel previousDataAccessLevel,
+      AgentType agentType) {
+    // In the event of an offline cronjob, the request may have been made anonymously and there is
```

Comment:
Maybe pull this preamble into its own helper, unless you're sure none of the other methods need to do this.

Reply:
They don't currently. I think this is a symptom of this method being too heavily overloaded so I'd rather leave the wart here than bless it in a utility class.